peer/peer.go

```diff
@@ -1406,7 +1422,7 @@ out:
 		// Read a message and stop the idle timer as soon as the read
 		// is done.  The timer is reset below for the next iteration if
 		// needed.
-		rmsg, buf, err := p.readMessage()
+		rmsg, buf, err := p.readMessage(p.wireEncoding)
```

Comment:
This is a race. It is reading a variable on the peer instance that is being set in a separate goroutine.